src/main/java/org/wso2/lsp4intellij/extensions/LSPExtensionManager.java

```diff
@@ -49,4 +53,28 @@
      * @param context The client context which can be used by the LanguageClient implementation.
      */
     LanguageClient getExtendedClientFor(ClientContext context);
+
+    /**
+     * Some language servers might only need to start for files which has a specific content. This method can be used
+     * in such situation to control whether the file must be connected to a language server which is registered for the
+     * extension of this file.
+     *
+     * <b>Note:</b> By default this method returns <code>true</code>
+     *
+     * @param file PsiFile which is about to connect to a language server.
+     * @return <code>true</code> if the file is supported.
+     */
+    default boolean isFileContentSupported(@NotNull PsiFile file) {
```

Comment:
I'm seeing that this util method is still not used. Could you please explain why do we need this here? :) (i.e. instead of somewhere like `FileUtils`?)

Reply:
I think this change is not part of this. I will fix it tonight and update the pr

Reply:
Sure :) thanks

Reply:
Done please merge this and let me know, i will then merge the label provider PR